genMarketplace.js

```diff
@@ -85,6 +85,91 @@ function createReadmeLink(listItem, itemType) {
   return Promise.resolve();
 }
 
+function travelDependenciesJson(firstLvlDepsJson, depsJson, startKey) {
+  /* Travels over the dependencies json to create a depdendencies map for each entry starting with startKey like so:
+  {
+    startKey: {
+      mandatory: {
+        packName: {
+          version: version
+        }
+      },
+      optional: {
+        packName: {
+          version: version
+        }
+      },
+      version: "...",
+      "mandatoryCount": ...,
+      "optionalCount": ...
+    }
+  }
+  */
+
+  if (startKey in depsJson === false) {
+    travelDependencies(depsJson, startKey, firstLvlDepsJson);
+    depsJson[startKey]['mandatoryCount'] = Object.keys(depsJson[startKey]['mandatory']).length;
+    depsJson[startKey]['optionalCount'] = Object.keys(depsJson[startKey]['optional']).length;
+  }
+}
+
+function travelDependencies(depsJson, startKey, firstLvlDepsJson) {
+  // Travel over the dependencies json of a given Pack dependency type, while collecting all sub-dependecy mandatory packs
+  
+  if (startKey in depsJson === false) {
```

Comment:
```suggestion
  if !(startKey in depsJson) {
```
Same here regarding the comment :)